desktop/Makefile.am

```diff
@@ -25,32 +25,107 @@
 # OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
 # ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-include $(top_srcdir)/mk/layout.am
-
-dist_desktop_DATA = fontforge.desktop
-dist_hicolor16x16_DATA = icons/16x16/apps/fontforge.png
-dist_hicolor22x22_DATA = icons/22x22/apps/fontforge.png
-dist_hicolor24x24_DATA = icons/24x24/apps/fontforge.png
-dist_hicolor32x32_DATA = icons/32x32/apps/fontforge.png
-dist_hicolor48x48_DATA = icons/48x48/apps/fontforge.png
-dist_hicolor64x64_DATA = icons/64x64/apps/fontforge.png
-dist_hicolor128x128_DATA = icons/128x128/apps/fontforge.png
-dist_hicolor256x256_DATA = icons/256x256/apps/fontforge.png
-dist_hicolorscalable_DATA = icons/scalable/apps/fontforge.svg
-dist_mimepackages_DATA = fontforge.xml
-
-install-data-hook: update-mime-database update-desktop-database
+ADDDESKTOP = "`$(TOUCH) $(DESTDIR)$(HICOLORDIR);`"
+
+if USE_THEME_TANGO
+ICON_SRC = $(srcdir)/tango
+endif
+
+if USE_THEME_2012
+ICON_SRC = $(srcdir)/2012
+endif
+
+desktopdir = $(PACKAGEROOTDIR)/applications
+desktop_DATA = fontforge.desktop
```

Comment:
Have you not changed the appid on purpose?
Otherwise these files, i.e. the desktop, appdata, and icon files, need to be renamed to resemble the appid, i.e. org.fontforge.FontForge.

Reply:
I'll need to review this in better detail when I get propper internet again.
This patch attempts to try meet adding a desktop (requested), and also attempt to maintain older ver1 information, but also try and include newer data in the metainfo directory to meet latest version requirements.
Do you think making a duplicate of fontforge.metainfo.xml in /usr{/local}/share/appdata could be helpful? I did see a couple metainfo.xml in the appdata directory too.